controller/engine_image_controller.go

```diff
@@ -393,73 +393,70 @@ func (ic *EngineImageController) syncNodeDeploymentMap(engineImage *longhorn.Eng
 }
 
 // handleAutoUpgradeEngineImageToDefaultEngineImage automatically upgrades volume's engine image to default engine image when it is applicable
-func (ic *EngineImageController) handleAutoUpgradeEngineImageToDefaultEngineImage() error {
-	concurrentAutomaticEngineUpgradePerNodeLimit, err := ic.ds.GetSettingAsInt(types.SettingNameConcurrentAutomaticEngineUpgradePerNodeLimit)
+func (ic *EngineImageController) handleAutoUpgradeEngineImageToDefaultEngineImage(currentProcessingImage string) error {
+	defaultEngineImage, err := ic.ds.GetSettingValueExisted(types.SettingNameDefaultEngineImage)
 	if err != nil {
 		return err
 	}
-	if concurrentAutomaticEngineUpgradePerNodeLimit <= 0 {
+
+	// To avoid multiple managers doing upgrade at the same time, only allow the
+	// manager that is responsible for the default engine image to do the upgrade
+	if currentProcessingImage != defaultEngineImage {
 		return nil
 	}
```

Comment:
This is more than just limiting to a single node, this limits to just events from the `defaultEngineImage`.
With the 30s resync period this will work for now, but might need to be looked at again later once we remove the 30s resync period.

Reply:
Yeah, this was the reason that I did the check `defaultEngineImage .Status.OwnerID == ec.ControllerID` in the previous push since it captures more events. But it still doesn't guarantee 100%. So, I switched to this recommended check to make the logic independent of the `Status.OwnerID` field.